### PR TITLE
[SM] FolderNav and Folder Page Updates

### PR DESCRIPTION
**Pull request by U-DON**
- Work on department-of-veterans-affairs/messaging-fe#5.
- Split non-default folders into their own sublist when there are more than 5 folders.
- Fixes for folder page; switching to another folder will update folder content.
- Placeholders for links to folder nav actions (create new folder, manage folders).
## Preview

screenshot from 2016-09-14 21 48 16

### CODE REVIEW

**Comment by U-DON** [comment]
Be sure to squash merge rather than create a merge commit when this is ready to be merged.

File: src/js/messaging/components/FolderNav.jsx [hunk 1]
```diff
@@ -43,10 +127,13 @@ FolderNav.propTypes = {
       /* eslint-disable */
       folder_id: React.PropTypes.number.isRequired,
```
Comment (webinista) [inline src/js/messaging/components/FolderNav.jsx]: Why snake case instead of camel case?

File: src/js/messaging/components/FolderNav.jsx [hunk 2]
```diff
@@ -1,40 +1,124 @@
...
+          {this.makeFolderLink(folder)}
+        </li>
+      );
     });
 
+    folderList.push(myFolders);
+
+    const folderActions = (
+      <li className="messaging-folder-nav-actions">
+        <a>
```
Comment (webinista) [inline src/js/messaging/components/FolderNav.jsx]: These should be `button` elements.

File: src/sass/messaging/partials/_messaging-app.scss [hunk 3]
```diff
@@ -14,10 +14,41 @@
...
+    line-height: 2rem;
+  }
+}
+
+.messaging-folder-nav-actions {
+  a {
+    color: $color-primary;
+    text-decoration: underline;
+
+    i {
```
Comment (webinista) [inline src/sass/messaging/partials/_messaging-app.scss]: This doesn't need to be nested inside of the `a` block. `.messaging-folder-nav-actions i` is specific enough.

**Review by webinista — CHANGES_REQUESTED** [review]

File: src/js/messaging/components/FolderNav.jsx (hunk 1 above)
Comment (U-DON) [inline src/js/messaging/components/FolderNav.jsx]: When we were using the mock Heroku API, the 'X-Key-Inflection' header didn't work, so I just went with the snake cased keys. I just tried the 'X-Key-Inflection' again on the staging API request, and it seemed to work, so I can change this in a separate branch.

**Review by webinista — APPROVED** [review]

**Comment by webinista** [comment]
LGTM once the tests pass

**Pull request merged**